pkg/cmd/admin/image/verify-signature.go

```diff
@@ -0,0 +1,220 @@
+}
+
+func NewCmdVerifyImageSignature(name, fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	opts := &VerifyImageSignatureOptions{ErrOut: errOut, Out: out}
+	cmd := &cobra.Command{
+		Use:     fmt.Sprintf("%s IMAGE [--confirm]", name),
+		Short:   "Verifies the given IMAGE signature with local public key",
+		Long:    verifyImageSignatureLongDesc,
+		Example: fmt.Sprintf(verifyImageSignatureExample, name),
+		Run: func(cmd *cobra.Command, args []string) {
+			kcmdutil.CheckErr(opts.Complete(f, cmd, args, out))
+			if opts.Remove {
+				kcmdutil.CheckErr(opts.removeImageSignature())
+			} else {
+				kcmdutil.CheckErr(opts.Run())
+			}
+		},
+	}
+
+	cmd.Flags().BoolVar(&opts.Confirm, "confirm", opts.Confirm, "If true, the result of the verification will be recorded to an image object.")
+	cmd.Flags().BoolVar(&opts.Remove, "remove", opts.Remove, "If set, the current signature verification will be removed from the image.")
+	cmd.Flags().StringVar(&opts.PublicKeyFilename, "public-key", opts.PublicKeyFilename, "A path to a public GPG key to be used for verification.")
+	return cmd
+}
+
+func (o *VerifyImageSignatureOptions) Complete(f *clientcmd.Factory, cmd *cobra.Command, args []string, out io.Writer) error {
+	if len(args) != 1 {
+		return kcmdutil.UsageError(cmd, "exactly one image must be specified")
+	}
+	o.InputImage = args[0]
+	var err error
+
+	// If --public-key is provided only this key will be used for verification and the
+	// .gnupg/pubring.gpg will be ignored.
+	if len(o.PublicKeyFilename) > 0 {
+		o.PublicKey, err = ioutil.ReadFile(o.PublicKeyFilename)
+		if err != nil {
+			return err
+		}
+	}
+	if o.Client, _, err = f.Clients(); err != nil {
+		return err
+	}
+	if o.Confirm && !o.Remove {
+		if me, err := o.Client.Users().Get("~"); err != nil {
+			return err
+		} else {
+			o.CurrentUser = me.Name
+		}
+	}
+
+	return nil
+}
+
+// verifySignature verifies the image signature and return the identity when the signature
+// is valid.
+// TODO: This should be calling the 'containers/image' library in future.
+func (o *VerifyImageSignatureOptions) verifySignature(signature []byte) (string, error) {
+	var (
+		mechanism SigningMechanism
+		err       error
+	)
+	// If public key is specified, use JUST that key for verification. Otherwise use all
+	// keys in local GPG public keyring.
+	if len(o.PublicKeyFilename) == 0 {
+		mechanism, err = newGPGSigningMechanismInDirectory("")
+	} else {
+		mechanism, _, err = newEphemeralGPGSigningMechanism(o.PublicKey)
+	}
+	if err != nil {
+		return "", err
+	}
+	defer mechanism.Close()
+	_, identity, err := mechanism.Verify(signature)
+	if err != nil {
+		return "", err
+	}
+	return string(identity), nil
+}
+
+// removeImageSignature removes the current image signature from the Image object by
+// erasing all signature fields that were previously set (when image signature was
+// previously verified).
+func (o *VerifyImageSignatureOptions) removeImageSignature() error {
```

Comment:
i think remove in this context should remove all verification or you should provide the signature name to remove (but I don't think we have to go that deep now)